x-pack/legacy/plugins/siem/server/lib/timeline/routes/import_timelines_route.ts

```diff
@@ -65,28 +68,34 @@ export const importTimelinesRoute = (
       },
     },
     async (context, request, response) => {
-      const siemResponse = buildSiemResponse(response);
-      const savedObjectsClient = context.core.savedObjects.client;
-      if (!savedObjectsClient) {
-        return siemResponse.error({ statusCode: 404 });
-      }
-      const { filename } = request.body.file.hapi;
+      try {
+        const siemResponse = buildSiemResponse(response);
+        const savedObjectsClient = context.core.savedObjects.client;
+        if (!savedObjectsClient) {
+          return siemResponse.error({ statusCode: 404 });
+        }
 
-      const fileExtension = extname(filename).toLowerCase();
+        const { file } = pipe(
+          importTimelinesPayloadSchema.decode(request.body),
+          fold(throwErrors(Boom.badRequest), identity)
+        );
 
-      if (fileExtension !== '.ndjson') {
-        return siemResponse.error({
-          statusCode: 400,
-          body: `Invalid file extension ${fileExtension}`,
-        });
-      }
+        const { filename } = file.hapi;
 
-      const objectLimit = config().get<number>('savedObjects.maxImportExportSize');
+        const fileExtension = extname(filename).toLowerCase();
+
+        if (fileExtension !== '.ndjson') {
+          return siemResponse.error({
+            statusCode: 400,
+            body: `Invalid file extension ${fileExtension}`,
+          });
+        }
+
+        const objectLimit = config().get<number>('savedObjects.maxImportExportSize');
 
-      try {
         const readStream = createTimelinesStreamFromNdJson(objectLimit);
         const parsedObjects = await createPromiseFromStreams<PromiseFromStreams[]>([
-          request.body.file,
+          (file as unknown) as Readable,
```

Comment:
I'm struggling with the `file` 's type here, guess there's better way to fix this.